### PR TITLE
Add circe source code to checksum library

### PR DESCRIPTION
**Pull request by rdhabalia**
### Motivation

```Pulsar-checksum``` uses ```circe``` library to compute checksum but it's release-binaries are not available. So, to avoid depending on snapshot version, import the source code of this library.

### Modifications

Import source classes from ```circe``` repo which requires to compute crc32c checksum.

### Result

No functionality change. However, while building pulsar, it doesn't have to download circe binaries.

### CODE REVIEW

**Comment by yahoocla** [comment]
CLA is valid!

File: NOTICE [hunk 1]
```diff
@@ -26,3 +26,12 @@ Google's data interchange format, which can be obtained at:
     * license/LICENSE.protobuf.txt (New BSD License)
   * HOMEPAGE:
     * http://code.google.com/p/protobuf/
+
+This product contains a modified portion of and depends on 'circe',
+Hash Algorithm Framework & Library, which can be obtained at:
+
+  * LICENSE:
+    * licenses/LICENSE-2.0 (Apache License)
```
Comment (rdhabalia) [inline NOTICE]: @merlimat : should we add any additional information under LICENSE.

File: NOTICE [hunk 2]
```diff
@@ -26,3 +26,12 @@ Google's data interchange format, which can be obtained at:
     * license/LICENSE.protobuf.txt (New BSD License)
```
Comment (merlimat) [inline NOTICE]: Actually.. this file we don't have it there :)

File: NOTICE (hunk 1 above)
Comment (merlimat) [inline NOTICE]: We can just mention : 

```
LICENSE : Apache License, Version 2.0
```

File: NOTICE [hunk 3]
```diff
@@ -26,3 +26,12 @@ Google's data interchange format, which can be obtained at:
...
   * HOMEPAGE:
     * http://code.google.com/p/protobuf/
+
+This product contains a modified portion of and depends on 'circe',
+Hash Algorithm Framework & Library, which can be obtained at:
+
+  * LICENSE:
+    * licenses/LICENSE-2.0 (Apache License)
+  * Git:
```
Comment (merlimat) [inline NOTICE]: HOMEPAGE

File: NOTICE (hunk 2 above)
Comment (rdhabalia) [inline NOTICE]: yes. So, should we change it to LICENSE : Google Inc. or we can keep it as it is.

**Comment by merlimat** [comment]
I think we should also import unit tests

File: NOTICE (hunk 2 above)
Comment (merlimat) [inline NOTICE]: We should add the protobuf license file as part of a separate PR

Comment (rdhabalia) [inline NOTICE]: I have added ```protobuf/LICENSE.protobuf.txt```

**Comment by rdhabalia** [comment]
imported unit-test cases after converting from junit to testng.

File: pulsar-checksum/pom.xml [hunk 4]
```diff
@@ -53,29 +53,10 @@
...
-      <artifactId>circe-common</artifactId>
-      <version>1.1-20140603.044903-1</version>
-    </dependency>
-
-    <dependency>
-      <groupId>com.scurrilous</groupId>
-      <artifactId>circe-crc</artifactId>
-      <version>1.1-20140603.044857-1</version>
+      <groupId>com.googlecode.jmockit</groupId>
+      <artifactId>jmockit</artifactId>
```
Comment (merlimat) [inline pulsar-checksum/pom.xml]: Shouldn't this be added to `test` scope?

Comment (rdhabalia) [inline pulsar-checksum/pom.xml]: yes.. i missed one commit to push.. it's only required for test so, scope should be test. made the change.

**Pull request merged**